framework/src/modules/token/commands/transfer_cross_chain.ts

```diff
@@ -141,7 +134,7 @@ export class TransferCrossChainCommand extends BaseCommand {
 		const escrowAccountKey = escrowStore.getKey(params.receivingChainID, params.tokenID);
 		const escrowAccoutExists = await escrowStore.has(context, escrowAccountKey);
 
-		if (tokenChainID.equals(this._ownChainID) && !escrowAccoutExists) {
+		if (tokenChainID.equals(context.chainID) && !escrowAccoutExists) {
```

Comment:
Comment about line 159: After subtracting the amount from sender's balance, we need to update the escrow in case  `tokenChainID.equals(context.chainID)` (see [`transferCrossChainInternal` fuction in LIP](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0051.md#transfercrosschaininternal)) before proceeding to emit event and interop.send.

It seems to be that this escrow update is missing here, am I missing sthg?

Reply:
In the `transferCrossChain` function (file [framework/src/modules/token/method.ts](https://github.com/LiskHQ/lisk-sdk/pull/8078/files#diff-a081182af4f79a82f8bffea268e7758024491c6236e36348c34bd9d35e2cb4c7) lines 460-463 this step is added, therefore I guess it is really missing here.

Perhaps implementing `transferCrossChainInternal` as in the LIP might help to avoid such inconsistencies.